src/Item.php

```diff
@@ -44,11 +44,18 @@ public function __construct(string $name)
 
     abstract public function getType(): string;
 
+    abstract public function canBeParentOfItem(Item $child): bool;
+
     public function getName(): string
     {
         return $this->name;
     }
 
+    public function isEqualName(string $name): bool
```

Comment:
I don't think that deserves a method. `->getName() === $name` should be enough.

Reply:
fixed